Write first article

hervit0 opened the pull request:
Main risks for me:
- a bit too long
- some parts might be too technical 🤔

hervit0 committed:
Write first article

hervit0 commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,112 @@
+
+With few words, the `Once upon a time` is: I grew up on a French island ([Reunion Island](https://www.google.co.uk/maps/place/Reunion/@-21.1351121,55.2471124,10z/data=!3m1!4b1!4m5!3m4!1s0x2178778110b8e43b:0x4a7f8e89ecdbeaf9!8m2!3d-21.115141!4d55.536384) - oversea territory), did a normal scholarship until my _baccalauréat_ (UK GCSE equivalent), attended to intensive courses in mathematics, physic ...
+
+From that point, the career path was almost sealed: I would have spent the next decades in the same company (because it shows loyalty and mind diversity is not really good you know?), painfully risen through the ranks, fought against this silly bureaucracy where old people teach you what to do in a world they can no longer understand, gone in retirement at 70 years old with nothing (stay lucid, t ...
```
`old people`, not politically correct

Fixed to `more senior people`:  3386394

hervit0 commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,112 @@
+As the software industry is moving fast regarding tools and frameworks, you constantly learn how to use a new technology, how to improve an existing one, how to deprecate another...
+On one side, it is pretty good for your brain, who likes to do the same thing every day for decades?
+On the other hand, it forces you to reconsider your pool of knowledge more often than you think. A framework that you master now, might die in 6 months. A programming language that you are fluent in, might be not used in 1 year time - a stronger one will soar and make your language useless or less efficient.
+It is quite harmful to lazy software engineers who are not curious and are not expressing any passion for their work: they simply rot with their deprecated pieces knowledge and slowly be out of the game. Don't be attached to syntax and frameworks (they die in a way or another): theories, concepts and principles will always prevail.
```
Flagged as too harsh

Fixed: 892cdc0

hervit0 commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,112 @@
+- you can not edit a file (or it is complicated) if someone has already opened it
+- you pick a file `my-file-v32.txt`, copy it, make some enhancements, rename the copy to `my-file-v33.txt`, store `v32` in a folder called `old` with the 31 other files
+- unless you have studied to Hogwarts, it is not possible to clearly see all the changes between `v32` and `v33`
+- you can easily mess up in the shared folder and lose 4 hours of your time with the documentation controller (yes, you are reading correctly, it is a job)
```
Flagged as too harsh

Fixed: 892cdc0

hervit0 commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,112 @@
+## A perpetual evolution
+
+When you work in an industry which is relying on fundamental sciences (chemistry, physics, mathematics), unless your industry is a very specific R&D one, you are often using old and solid principles, which have been discovered over the past 500 years, sometimes more for some core mathematics concepts.
+Therefore, designing a chemical plant roughly consists in applying mature scientific principles. All the beauty of this kind of engineering is to know whenever using the appropriate theorem or the correct formula.
```
Nuance to add about reliability in the chemical industry
Fixed: 892cdc0

hervit0 commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,112 @@
+
+Back to the chemical industry, the uncertainty on the ability of a machine to work or not, fosters over-design: your formula says `5 kW`, but you will not take any risk and make it `10 kW`, costing you extra-money.
+For the previous line of code, executing it on my machine will give me an instant feedback on the validity and the correctness of it. But we go way more beyond: how to make sure it works on another machine? Do I have to run it on my machine each time I need to prove that it still work?
+I introduce here a fundamental principle in my job: testing. It is a wide, significant subject, but here are the main takeaways. The line of code above can be naively tested as `assert String.upcase("bonjour") == "BONJOUR"`: if it is correct it returns a positive response, otherwise it fails. We will encapsulate this assertion within a script with other assertions and run all our scripts whenever ...
```
Nuance with how to make a good reliant plant.

Fixed: 892cdc0

emilywoods commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,114 @@
+# From chemical engineering to software development
+
+1st of April, 2016, Paris, _Gare du Nord_. With my backpack and my monitor in a bag, I onboard a Eurostar train to London with my cousin Tom and his wife Emma. I leave a chemical engineer position behind me, for a software developer one.
```
`I leave a chemical` - I think this should be `was leaving` or `left` :)

hervit0 replied:
oh yeah you're right 🙈

emilywoods commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,114 @@
+
+1st of April, 2016, Paris, _Gare du Nord_. With my backpack and my monitor in a bag, I onboard a Eurostar train to London with my cousin Tom and his wife Emma. I leave a chemical engineer position behind me, for a software developer one.
+
+The year is now 2018, I work for a company called [Driftrock](https://www.driftrock.com/): we boost the marketing power of our clients through social media. I'm a software developer there, taking care of an awesome microservices fleet with an heavily-skilled team and juggling between back-end, front-end and DevOps, using various programming languages to achieve our goals. Before that, I have spen ...
```
- `a heavily-skilled`
- `Before that, I had`
-  I think this sentence could be broken up like: `Before that, I had spent more than a year at [The App Business](http://www.theappbusiness.com/), a tech company with a specialisation in mobile technologies. There I was a junior software engineer, learning key concepts like Agile, testing and clean code.`  What do you think?

hervit0 replied:
Hehe, thanks for the particular check for TAB 😏 ! I'm more than happy with your adjustment 😄

emilywoods commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,114 @@
+
+The year is now 2018, I work for a company called [Driftrock](https://www.driftrock.com/): we boost the marketing power of our clients through social media. I'm a software developer there, taking care of an awesome microservices fleet with an heavily-skilled team and juggling between back-end, front-end and DevOps, using various programming languages to achieve our goals. Before that, I have spen ...
+
+Time for a check-up after 2 years, isn't it?
```
`check-in` might be more appropriate

emilywoods commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,114 @@
+
+Unfortunately, I give here only the differences between a non-tech industry and the software industry, therefore, the purpose of this article is not about how to start your software engineer career (however, you can still drop me a message if you are looking for any pieces of advice).
+
+The differences between the British and the French way of working are not tackled here, it is a wide subject deserving a full and proper article (trailer: I am writing from London, which gives you a good idea of my humble opinion).
```
I don't think the part in brackets is necessary i.e. `(trailer:..`

hervit0 replied:
Hm also too caustic towards my people 😂 , better to remove it.

For the Queen 🇬🇧 !

emilywoods commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,114 @@
+
+With few words, the `Once upon a time` is: I grew up on a French island ([Reunion Island](https://www.google.co.uk/maps/place/Reunion/@-21.1351121,55.2471124,10z/data=!3m1!4b1!4m5!3m4!1s0x2178778110b8e43b:0x4a7f8e89ecdbeaf9!8m2!3d-21.115141!4d55.536384) - oversea territory), did a normal scholarship until my _baccalauréat_ (UK GCSE equivalent), attended to intensive courses in mathematics, physic ...
+
+From that point, the career path was almost sealed: I would have spent the next decades in the same company (because it shows loyalty and mind diversity is not really good you know?), painfully risen through the ranks, fought against this silly bureaucracy where more senior people teach you what to do in a world they can no longer understand, gone in retirement at 70 years old with nothing (stay  ...
```
- 'mind diversity is not really good you know' - do you have an example of this? it comes across as quite subjective. From my experience, I don't think the problem is a discouragement of broadening your mind, but often it's less of a necessity in chem eng, given that you're often dealing with well-established techniques and processes, and usually there's a tried and tested way of getting things done in order to meet whatever health and safety, technical, quality requirements, and hence there's not much need for divergence...
- 'where more senior people teach you what to do in a world they can no longer understand' - woah! bit harsh :P perhaps something more along the lines of: 'where seniority often results in adopting a more managerial role and becoming removed from technical processes' and perhaps 'and hence become disconnected from technological advances.' However you want to write it, I think this should be toned-down or elaborated on to explain why you think like this!

hervit0 replied:
- for the first point, I reckon it was too curt, the point was more about bringing contrasts with the software industry where we are encouraged to grab the best from everything (e.g. pattern from a language to another, or more specifically, bring parametrized tests idea from C# to Java - rings a bell 😏 ?) and then forge a wide-opened mindset. But indeed, it can be misunderstood here, your point is totally valid, the chem engineering is in no way forbidding you to extend your reasoning
- Ahaha 😂 I think everyone said something on this one, I admit that I had one of my former managers in mind! Your sentences are way more politically acceptable, thanks a ton 😄 !

emilywoods commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,114 @@
+
+Fortunately enough, I did not spend decades where I started. After only 2 years and a half in Paris, doing the same repetitive and boring tasks, under an awful and stressful management, I was on my knees, dreamless and buried below this dull routine under morphine.
+
+After a lot of discussions with my cousin [Tom](https://github.com/ostapneko), his wife [Emma](https://github.com/emma-p), and her brother [Alex](https://github.com/onima) (all of them are developers), I decided to make the move. I will be forever thankful to them for having the patience and taking the time to onboard me in this amazing industry.
```
🙌

emilywoods commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,114 @@
+
+## Timescale and feedback loop are not the same
+
+Let's start with the most obvious one: velocity of execution is nothing but comparable.
```
this could be more simply written with just: `Let's start with the most obvious one: velocity of execution.`

emilywoods commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,114 @@
+
+## A perpetual evolution
+
+When you work in an industry which is relying on fundamental sciences (chemistry, physics, mathematics), unless your industry is a very specific R&D one, you are often using old and solid principles, which have been discovered over the past 500 years, sometimes more for some core mathematics concepts.
```
I prefer 'well-established' to 'old'

hervit0 replied:
👍 !

Did you get the pun with the solid principles 😂 ?

emilywoods commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,114 @@
+Of course, I only describe (maybe with a little exaggeration) the paths for a chemical engineer in an engineering office context, chemical engineer on production fields (e.g. offshore platforms) is something very different.
+
+Things are shaped differently in the software industry. As long as you can write a line of code, you can become: software developer, data scientist/analyst, automation/manual tester, DevOps engineer, embedded software engineer, software architect, implementation engineer...
+Furthermore, upgrading your role level is pretty fair and is based on your skills, rather than being based on non-scientific parameters like the number of years spent with the role of "chief manager of a team composed of me and myself", or how grey my hair is (or no hair at all). If someone has all the required potential to reach the Moon, just let her/him go.
```
'upgrading your role level is pretty fair' - I would probably add 'usually' in here. I think we're both lucky with the companies we've encountered, and no doubt there are still plenty of software-based companies that don't always hire and promote based on these factors.

hervit0 replied:
Arg, you've got a damn fair point here!

emilywoods commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,114 @@
+
+## Side projects and own experimentations
+
+Sadly, you cannot assemble your own nuclear reactor in your garden as a weekend hobby (or you would be in trouble, wouldn't you?). Or you cannot mount an industrial methanol pump in your garage. Difficult to make your husband/wife digest: "Just to see if it works".
```
could just write 'partner' instead of 'husband/wife' :)

emilywoods commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,114 @@
+
+There are still billions of things to say, but I will wrap up this article with these points:
+- the community is awesome and always aims towards excellence: I have stopped counting meetups, conventions, conferences that I have been to during these past 2 years. But I can give you for sure the number of chemical engineering events I attended during 2 years and a half: 0.
+- the majority of our technologies is open-source, meaning that you can always contribute to the latest and advanced framework if you want. Money for sure is a powerful motivation engine, but passion is the most rewarding one.
```
'our technologies are open-source'

hervit0 replied:
🙈 ... thanks!!

emilywoods reviewed (commented):
🙌 Really well-written, and quite relatable ;)  I added a few comments, let me know what you think!

hervit0 committed:
Polish things with the awesome feedback from Miss Woods :ok_hand:

hervit0 reviewed (commented):
@emilywoods many thanks for your review!

You've spotted some damn good points there 👌I didn't disagree on any of them, some nuances were required here and there, you got them right!

Also thanks for the left-over typo issues 🙈 ...

Update -> ca0a61d

hervit0 commented:
arg sorry @emilywoods, ~I've committed and then submitted the response-review 🙈 therefore, my comments are dissociated from yours 🙈 ...~ Everything is alright on your review!

emilywoods commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,114 @@
+# From chemical engineering to software development
+
+1st of April, 2016, Paris, _Gare du Nord_. With my backpack and my monitor in a bag, I onboard a Eurostar train to London with my cousin Tom and his wife Emma. I left a chemical engineer position behind me, for a software developer one.
```
chemical engineering

hervit0 replied:
Arg, a left-behind 🙈 , cheers: 3360e00

emilywoods reviewed (commented):
👌 awesome. I also thought of another point you could make when drawing comparisons between the industries. Since many of the techniques used in software have their origins or are also used in manufacturing (e.g. lean, kanban) you could mention that (maybe in the conclusion?), and suggest that the manufacturing industry could also learn from other techniques used in software

hervit0 commented:
@emilywoods good shout for the techniques!

Hm I was about to stick it where I'm talking about the Agile methodology, but it didn't fit well and this paragraph was already big enough. So I've followed your piece of advice and stick it into the conclusion (I should rename `wrap up` in `conclusion`, shouldn't I 😂 ?) 

I modified a bit your idea, and generalize it, lemme know if it still makes sense 🤔 .

Also, I've added something at the very bottom, let me know if I'm allowed to 😄 !

Update => 3360e00

emilywoods commented on _posts/2018-04-01-from-chemical-engineering-to-software-development.md:
```diff
@@ -0,0 +1,117 @@
+
+With few words, the `Once upon a time` is: I grew up on a French island ([Reunion Island](https://www.google.co.uk/maps/place/Reunion/@-21.1351121,55.2471124,10z/data=!3m1!4b1!4m5!3m4!1s0x2178778110b8e43b:0x4a7f8e89ecdbeaf9!8m2!3d-21.115141!4d55.536384) - oversea territory), did a normal scholarship until my _baccalauréat_ (UK GCSE equivalent), attended to intensive courses in mathematics, physic ...
+
+From that point, the career path was almost sealed: I would have spent the next decades in the same company, painfully risen through the ranks, fought against this silly bureaucracy where seniority often results in adopting a more managerial role and becoming removed from technical processes - hence become disconnected from technological advances, and then gone in retirement at 70 years old with  ...
```
Just one more small correction!
- forgot 'ing' at the end of 'become' i.e `becoming more disconnected from technological advances` (or tbh you could remove this last part entirely)
- and possibly you could also remove 'with nothing' :)

hervit0 replied:
Ah thanks @emilywoods I've lightened a bit 🦅 : 44e0279

Just to let you know, a friend of mine was reading, and asked this very interesting question:
> I am curious, did you find it hard learning to write code and understand all the tech jargon or was it already an interest therefore quite easy to pick up?

That's a good damn question for both us, but I'm afraid it's off-topic here 🤔
